src/components/organisms/code/addModal.tsx

```diff
@@ -84,10 +80,12 @@ export const AddFileModal = () => {
 				namespaces.projectUICode,
 				t("fileAddFailedExtended", { fileName: name, projectId, error })
 			);
+		} finally {
+			setIsCreatingFile(false);
+			clearErrors();
+			closeModal(ModalName.addCodeAssets);
+			reset({ extension: { label: defalutFileExtension, value: defalutFileExtension }, name: "" });
```

Comment:
I think we better close the modal after the reset, in order to avoid a condition race.